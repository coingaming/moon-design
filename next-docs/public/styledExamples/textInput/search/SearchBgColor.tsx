import React from 'react';
import { TextInput } from '@heathmont/moon-core';

const Example = () => (
  <div className="flex flex-col lg:flex-row justify-around items-end w-full gap-2">
    <TextInput
      type="search"
      inputSize="medium"
      label="medium"
      placeholder="e.g. EPL"
      hintText="Informative message holder"
      backgroundColor="goku"
    />
    <TextInput
      type="search"
      inputSize="large"
      label="large"
      placeholder="e.g. EPL"
      hintText="Informative message holder"
      backgroundColor="goku"
    />
    <TextInput
      type="search"
      inputSize="xlarge"
      label="xlarge"
      placeholder="e.g. EPL"
      hintText="Informative message holder"
      backgroundColor="goku"
    />
  </div>
);

export default Example;
