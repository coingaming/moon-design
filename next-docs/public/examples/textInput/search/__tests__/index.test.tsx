/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { moonDesignLight, ThemeProvider } from '@heathmont/moon-themes';
import SearchWithLabel from '../SearchWithLabel';
import SearchWithoutLabel from '../SearchWithoutLabel';
import SearchWithHint from '../SearchWithHint';
import SearchDisabled from '../SearchDisabled';
import SearchError from '../SearchError';
import SearchBgColor from '../SearchBgColor';
import SearchReadonly from '../SearchReadonly';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>{component}</ThemeProvider>
);

describe('TextInput: type Search', () => {
  it('renders with label', () => {
    const testRenderer = renderer.create(renderWithTheme(<SearchWithLabel />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders without label', () => {
    const testRenderer = renderer.create(
      renderWithTheme(<SearchWithoutLabel />)
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders with hint', () => {
    const testRenderer = renderer.create(renderWithTheme(<SearchWithHint />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders disabled', () => {
    const testRenderer = renderer.create(renderWithTheme(<SearchDisabled />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders error', () => {
    const testRenderer = renderer.create(renderWithTheme(<SearchError />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('supports background colour', () => {
    const testRenderer = renderer.create(renderWithTheme(<SearchBgColor />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders readOnly', () => {
    const testRenderer = renderer.create(renderWithTheme(<SearchReadonly />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
