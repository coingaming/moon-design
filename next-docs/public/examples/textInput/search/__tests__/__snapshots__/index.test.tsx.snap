// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput: type Search renders disabled 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 bRHlks"
        disabled={true}
        placeholder=" "
        type="search"
        value="Text"
      />
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 cIZOPb"
        disabled={true}
        placeholder=" "
        type="search"
        value="Text"
      />
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 gBLEES"
      disabled={true}
    >
      <div
        className="Inner-sc-1soq84i-0 XqqD"
      >
        <input
          className="Input-qkdgvx-0 KFdrt"
          disabled={true}
          placeholder=" "
          type="search"
          value="Text"
        />
        <label
          className="LabelInner-vrrjt0-0 foFqgl"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Search renders error 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 ekGwYi"
        placeholder="e.g. EPL"
        type="search"
      />
      <p
        className="HintText-sc-1e9hzxr-0 jpqMLI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 erUZHR"
        placeholder="e.g. EPL"
        type="search"
      />
      <p
        className="HintText-sc-1e9hzxr-0 jpqMLI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 XqqD"
      >
        <input
          className="Input-qkdgvx-0 Rinj"
          placeholder="e.g. EPL"
          type="search"
        />
        <label
          className="LabelInner-vrrjt0-0 foFqgl"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 jpqMLI"
        role="alert"
      >
        Error message
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Search renders readOnly 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 bRHlks"
        placeholder=" "
        readOnly={true}
        type="search"
        value="Read only text"
      />
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 cIZOPb"
        placeholder=" "
        readOnly={true}
        type="search"
        value="Read only text"
      />
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 XqqD"
      >
        <input
          className="Input-qkdgvx-0 KFdrt"
          placeholder=" "
          readOnly={true}
          type="search"
          value="Read only text"
        />
        <label
          className="LabelInner-vrrjt0-0 foFqgl"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Search renders with hint 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 bRHlks"
        placeholder="e.g. EPL"
        type="search"
      />
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 cIZOPb"
        placeholder="e.g. EPL"
        type="search"
      />
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 XqqD"
      >
        <input
          className="Input-qkdgvx-0 KFdrt"
          placeholder="e.g. EPL"
          type="search"
        />
        <label
          className="LabelInner-vrrjt0-0 foFqgl"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Search renders with label 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 bRHlks"
        placeholder="e.g. EPL"
        type="search"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 cIZOPb"
        placeholder="e.g. EPL"
        type="search"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 XqqD"
      >
        <input
          className="Input-qkdgvx-0 KFdrt"
          placeholder="e.g. EPL"
          type="search"
        />
        <label
          className="LabelInner-vrrjt0-0 foFqgl"
        >
          xlarge
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Search renders without label 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <input
        className="Input-qkdgvx-0 bRHlks"
        placeholder="medium"
        type="search"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <input
        className="Input-qkdgvx-0 cIZOPb"
        placeholder="large"
        type="search"
      />
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 XqqD"
      >
        <input
          className="Input-qkdgvx-0 hnRIqU"
          placeholder="xlarge"
          type="search"
        />
        <label
          className="LabelInner-vrrjt0-0 foFqgl"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Search supports background colour 1`] = `
<div
  className="flex gap-4 items-end"
>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        medium
      </label>
      <input
        className="Input-qkdgvx-0 hZkrIH"
        placeholder="e.g. EPL"
        type="search"
      />
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <label
        className="Label-sc-1rdea5p-0 bJkxLR"
      >
        large
      </label>
      <input
        className="Input-qkdgvx-0 hQSrPI"
        placeholder="e.g. EPL"
        type="search"
      />
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
  <div
    className="m-4"
  >
    <div
      className="Container-sc-1f49upb-0 fVKDpE"
    >
      <div
        className="Inner-sc-1soq84i-0 kEpyYM"
      >
        <input
          className="Input-qkdgvx-0 KFdrt"
          placeholder="e.g. EPL"
          type="search"
        />
        <label
          className="LabelInner-vrrjt0-0 foFqgl"
        >
          xlarge
        </label>
      </div>
      <p
        className="HintText-sc-1e9hzxr-0 kLWKCo"
        role="alert"
      >
        Informative message holder
      </p>
    </div>
  </div>
</div>
`;
